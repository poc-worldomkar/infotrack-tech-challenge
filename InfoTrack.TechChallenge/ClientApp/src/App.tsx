import * as React from "react";
import { Route } from "react-router";
import DefaultLayout from "./layouts/DefaultLayout";
import Home from "./pages/Home";
import Configuration from "./pages/Configuration";

import "./custom.css";

export default () => (
  <DefaultLayout>
    <Route exact path="/" component={Home} />
    <Route path="/configuration" component={Configuration} />
  </DefaultLayout>
);
