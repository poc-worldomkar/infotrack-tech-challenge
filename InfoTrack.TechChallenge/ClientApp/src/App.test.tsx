import * as React from 'react';
// import * as ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import App from './App';

it('InfoTrack Seo Check App renders without crashing', () => {
  const storeFake = (state: any) => ({
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => ({ ...state }),
  });
  const store = storeFake({
    configuration: {
      searchEngines: [],
    },
  }) as any;
  const tree = renderer
    .create(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
  //   ReactDOM.render(
  //     <Provider store={store}>
  //       <MemoryRouter>
  //         <App />
  //       </MemoryRouter>
  //     </Provider>,
  //     document.createElement('div')
  //   );
});
