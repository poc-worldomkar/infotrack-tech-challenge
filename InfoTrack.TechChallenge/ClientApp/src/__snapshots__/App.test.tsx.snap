// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoTrack Seo Check App renders without crashing 1`] = `
<header
  className="makeStyles-root-1"
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary makeStyles-appBar-2 mui-fixed MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-4 MuiIconButton-colorInherit MuiIconButton-edgeStart"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
      </button>
      <h1
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap"
      >
        InfoTrack Tech Challenge
      </h1>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-6 makeStyles-drawerClose-8"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerClose-8 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="makeStyles-toolbar-9"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </span>
        </button>
      </div>
      <hr
        className="MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Home
            </span>
          </div>
        </div>
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemIcon-root"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
              />
            </svg>
          </div>
          <div
            className="MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              Configuration
            </span>
          </div>
        </div>
      </ul>
    </div>
  </div>
  <main
    className="makeStyles-content-10"
  >
    <div
      className="makeStyles-toolbar-9"
    />
    <div
      className="MuiGrid-root makeStyles-container-12 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column"
    >
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <h2
          className="MuiTypography-root MuiTypography-h6"
        >
          InfoTrack Seo Index Check
        </h2>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
            data-shrink={true}
            htmlFor="query"
            id="query-label"
          >
            Query
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="query"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              type="text"
              value="online title search"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-16 PrivateNotchedOutline-legendNotched-17"
              >
                <span>
                  Query
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <div
          className="MuiFormControl-root makeStyles-formControl-11 MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink={false}
            id="search-engine-label"
          >
            Search Engine
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
            onClick={[Function]}
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="search-engine-label search-engine"
              className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="search-engine"
              onBlur={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onMouseDown={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </div>
            <input
              aria-hidden={true}
              className="MuiSelect-nativeInput"
              onAnimationStart={[Function]}
              onChange={[Function]}
              required={false}
              tabIndex={-1}
              value=""
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-14 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-16"
              >
                <span>
                  Search Engine
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <label
          className="MuiFormControlLabel-root"
        >
          <span
            aria-disabled={false}
            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-18 MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-checked-19 Mui-checked MuiIconButton-colorPrimary"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={null}
          >
            <span
              className="MuiIconButton-label"
            >
              <input
                checked={true}
                className="PrivateSwitchBase-input-21"
                data-indeterminate={false}
                name="useStaticPages"
                onChange={[Function]}
                type="checkbox"
              />
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
          </span>
          <span
            className="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            Use static pages
          </span>
        </label>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Check SEO Index
          </span>
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item"
      />
    </div>
  </main>
</header>
`;
